core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -621,14 +622,18 @@ public void close() {
     public static abstract class Operation {
         private final Term uid;
         private long version;
+        private long seqNo;
         private final VersionType versionType;
         private final Origin origin;
         private Translog.Location location;
         private final long startTime;
         private long endTime;
 
-        public Operation(Term uid, long version, VersionType versionType, Origin origin, long startTime) {
+        public Operation(Term uid, long seqNo, long version, VersionType versionType, Origin origin, long startTime) {
```

Comment:
yeah. I felt it's safer this way as it makes people think about it. I was worried that we will call this constructor somewhere in production code. We can solve it through package private and utility test methods if we feel the need. Felt it's an overkill for now.